tools/thumbnaild-video/libavthumbnailer.cpp

```diff
@@ -204,6 +204,9 @@ QImage createVideoThumbnail(const QString &fileName, const QSize &requestedSize,
             } else if (ret != AVERROR(EAGAIN)) {
                 av_packet_unref(&packet);
                 return image;
+            } else {
+                // When EAGAIN is returned codec needs more data
+                break;
```

Comment:
Is it ok that `av_packet_unref(&packet)` isn't being done in this case? And is `thumbnailer.frame` guaranteed to be pointing to a valid frame?

Reply:
The av_packet_unref(&packet) is called a few lines later at end of the outer loop. In the case where the inner loop breaks here thumbnailer.frame is not set yet as next frame will be attempted during next iteration of the outer loop.

Reply:
ok, sounds sane!